### PR TITLE
chore(cli): log SDK calls when using `-vvv`

### LINKED ISSUE #32306: CLI: SDK call logging doesn't work anymore since SDKv3
### Describe the bug

Compare these 2 logs:

```
>> v2.166.0
[13:59:43] Assuming role 'arn:aws:iam::012345678901:role/cdk-REDACTED-lookup-role-012345678901-ap-northeast-1'.
[13:59:43] [trace]   SDK#forceCredentialRetrieval()
[13:59:43] [trace] SDK#ssm()
[13:59:43] [trace]   SDK#wrapServiceErrorHandling()
[13:59:43] [AWS ssm 200 0.296s 0 retries] getParameter({ Name: '/cdk-bootstrap/REDACTED/version' })
[13:59:43] [trace] SDK#cloudFormation()
[13:59:43] [trace]   SDK#wrapServiceErrorHandling()
[13:59:44] [AWS cloudformation 200 0.209s 0 retries] describeStacks({ StackName: 'REDACTED' })
[13:59:44] [AWS cloudformation 200 0.122s 0 retries] getTemplate({ StackName: 'REDACTED', TemplateStage: 'Original' })
[13:59:44] [trace] SdkProvider#resolveEnvironment()

>> v2.171.1-rc.0-16d61b1
[17:36:40] Assuming role 'arn:aws:iam::REDACTED:role/cdk-REDACTED-lookup-role-REDACTED-ap-northeast-1'.
[17:36:46] [trace] SDK#ssm()
[17:36:56] [trace] SDK#cloudFormation()
[17:37:09] [trace] SdkProvider#resolveEnvironment()
```

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Last Known Working CDK Version

_No response_

### Expected Behavior

-

### Current Behavior

-

### Reproduction Steps

-

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

-

### Framework Version

_No response_

### Node.js Version

-

### OS

-

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by rix0rrr**
(This work came out of the proxy issue research)

The logging of what SDK calls were being performed was lost during the migration of SDKv2 -> SDKv3.  Add it back.

Also in this PR:

- Set a timeout on `npm view`; if for network isolation reasons NPM can't connect to the server, it will make the CLI hang for a minute.
- Work around an issue where the CLI entirely fails if it is run as a user that doesn't have a home directory.

Closes #32306.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by rix0rrr** [comment]
Still a Draft because I'm not happy with how the messages are being formatted; there is too much detail in them as they include the entire API call payloads and responses (including somewhat-sensitive information).

But creating a PR already anyway because I want to store this code somewhere safe.

File: packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import { inspect } from 'util';
+import { Logger } from '@smithy/types';
+import { trace } from '../../logging';
+
+export class SdkToCliLogger implements Logger {
+  public trace(...content: any[]) {
+    // This is too much detail for our logs
+    // trace('[SDK trace] %s', this.fmtContent(content));
+    Array.isArray(content);
```
Comment (otaviomacedo) [inline packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts:9]: I don't understand this line.

File: packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+export function formatApiCall(content: any): string | undefined {
+  if (!isSdkApiCallSuccess(content) && !isSdkApiCallError(content)) {
+    return undefined;
+  }
+
+  const service = content.clientName.replace(/Client$/, '');
+  const api = content.commandName.replace(/Command$/, '');
+
+  const parts = [];
+  if (content.metadata.attempts > 1) {
```
Comment (otaviomacedo) [inline packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts:106]: Is `metadata` guaranteed to exist?

File: packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  /**
+   * This can be anything.
+   *
+   * For debug, it seems to be mostly strings.
+   * For info, it seems to be objects.
+   *
+   * Stringify and join without separator.
+   */
+  private fmtContent(content: any[]) {
+    if (content.length === 1) {
```
Comment (otaviomacedo) [inline packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts:87]: Can `content` not be an array with more than one error, for example?

File: packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts:106]: I think so but who knows.

File: packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts (hunk 3 above)
Comment (rix0rrr) [reply packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts:87]: If so we won't do the wrong thing, we just won't be doing a nice thing.

We'll see that in the logs at some point and fix it then.

File: packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/aws-cdk/lib/api/aws-auth/sdk-logger.ts:106]: I'll code more defensively

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**